rust/basic_ethereum/README.md

```diff
@@ -0,0 +1,179 @@
+      See [signing messages](https://internetcomputer.org/docs/current/developer-docs/smart-contracts/encryption/signing-messages#signing-messages-1)
+      for more details.
+
+If successful, you should see an output that looks like this:
+
+```bash
+Deploying: basic_ethereum
+Building canisters...
+...
+Deployed canisters.
+URLs:
+Candid:
+    basic_ethereum: https://a4gq6-oaaaa-aaaab-qaa4q-cai.raw.icp0.io/?id=<YOUR-CANISTER-ID>
+```
+
+Your canister is live and ready to use! You can interact with it using either the command line or using the Candid UI,
+which is the link you see in the output above.
+
+In the output above, to see the Candid Web UI for your ethereum canister, you would use the
+URL `https://a4gq6-oaaaa-aaaab-qaa4q-cai.raw.icp0.io/?id=<YOUR-CANISTER-ID>`. You should see the methods specified in the Candid file `basic_ethereum.did`.
+
+## Step 2: Generating an Ethereum address
+
+An Ethereum address can be derived from an ECDSA public key. To derive a user's specific address, identified on the IC
+by a principal, the canister uses its own threshold ECDSA public key to derive deterministically a new public key for
+each requested principal. To retrieve your Ethereum address, you can call the `ethereum_address` method on the
+previously deployed canister:
+
+```shell
+dfx canister --network ic call basic_ethereum ethereum_address
+```
+
+This will return an Ethereum address such as `("0x378a452B20d1f06008C06c581b1656BdC5313c0C")` that is unique to your
+principal. Your address will be different. You can view such address on any Ethereum block explorer such as Etherscan.
+
+If you want to send some ETH to someone else, you can also use the above method to enquiry about their Ethereum address
+given their IC principal
+
+```shell
+dfx canister --network ic call basic_ethereum ethereum_address '(opt principal "hkroy-sm7vs-yyjs7-ekppe-qqnwx-hm4zf-n7ybs-titsi-k6e3k-ucuiu-uqe")'
+```
+
+This will return a different Ethereum address as the one above, such
+as `("0x8d68f7B3cdb40A2E77071077658b01A9EA4B040F")`.
+
+## Step 3: Receiving ETH
+
+Now that you have your Ethereum address, let us send some (Sepolia) ETH to it:
+
+1. Get some Sepolia ETH if you don't have any. You can for example use such
+   a [faucet](https://www.alchemy.com/faucets/ethereum-sepolia).
+2. Send some Sepolia ETH to the address you obtained in the previous step. You can use any Ethereum wallet (e.g.,
+   Metamask) to do so.
+
+Once the transaction has at least one confirmation, which can take a few seconds,
+you'll be able to see it in your Ethereum address's balance, which should be visible in an Ethereum block explorer,
+e.g. https://sepolia.etherscan.io/address/0x378a452b20d1f06008c06c581b1656bdc5313c0c.
+
+## Step 4: Sending ETH
+
+You can send ETH using the `send_eth` endpoint on your canister, specifying an Ethereum destination address and an
+amount in the smallest unit (Wei). For example, to send 1 Wei to `0xdd2851Cdd40aE6536831558DD46db62fAc7A844d`:
```

Comment:
1 Wei?!?
The amount below doesn't match this (really) small value.

Reply:
I had forgotten this comment sorry. I changed the amount to match the 1 Wei [here](https://github.com/dfinity/examples/pull/931/commits/ff57fd030ffd422a0c858380c04c3ee573b4538a).